File: api/schemas/jobs.py
"""Generation job API schemas."""

from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, ConfigDict


class GenerationJobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    parent_job_id: Optional[str] = None
    job_type: str
    provider: str
    status: str
    progress: float
    external_job_id: Optional[str]
    workflow_id: Optional[str]
    model_name: Optional[str]
    params_json: dict[str, Any]
    result_json: dict[str, Any]
    error_message: Optional[str]
    created_at: datetime
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    progress_stage: Optional[str] = None
    progress_message: Optional[str] = None

File: frontend/src/types/jobs.ts
export type GlobalJobStatus = "pending" | "running" | "completed" | "failed" | "cancelled";
export type GlobalJobType = "script_generation" | "storyboard_generation" | "video_generation" | "research";

export interface GlobalJob {
  id: string;
  project_id: string;
  parent_job_id?: string | null;
  job_type: GlobalJobType;
  provider: string;
  status: GlobalJobStatus;
  progress: number;
  external_job_id: string | null;
  workflow_id?: string | null;
  model_name?: string | null;
  params_json: Record<string, unknown>;
  result_json: Record<string, unknown>;
  error_message: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
  progress_stage?: string | null;
  progress_message?: string | null;
  progress_current_scene?: number | null;
  progress_total_scenes?: number | null;
}

export interface JobAcceptedResponse {
  job_id: string;
}
